Sources/muterCore/Extensions/StringExtensions.swift

```diff
@@ -80,3 +80,36 @@ extension String {
         return String(self[range])
     }
 }
+
+extension String {
+    /// Converts a UTF-8 byte offset to a character offset in the given string.
+    ///
+    /// - Parameters:
+    ///   - utf8Offset: The UTF-8 byte offset to be converted.
+    /// - Returns: The corresponding character offset if the conversion is successful, or `nil` if can't convert to corresponding char offset.
+    ///
+    /// - Complexity: O(*n*), where *n* is the number of characters in the string.
+    func convertToCharOffset(from utf8Offset: Int) -> Int? {
+        guard utf8Offset >= 0 && utf8Offset <= utf8.count else {
+            return nil
```

Comment:
lets return the original `utf8Offset` to the caller, this way we don't introduce any regressions here: 

```swift
guard let nodePositionOffset = codeBlockDescription.convertToCharOffset(from: utf8Offset) else {
            return super.transform(node: node, mutatedSyntax: mutatedSyntax, at: mutationRange)
        }
```

Reply:
@rakaramos 
Is there something wrong with the call to `super.transform` ? Cuz eventually we will call `super.transform` anyway.   

If we return the original `utf8Offset`, there may be chance the crash will occur due to `indexOutOfRange`.
Can I update the logic to this?

```swift
guard let nodePositionOffset = codeBlockDescription.convertToCharOffset(from: utf8Offset) else {
    return node.codeBlockItemListSyntax
}
```

Reply:
nah, its fine, lets keep as you did before and keep an eye on it 👍 
this operator is very complex already